### PR TITLE
fix(invites): make email delivery logs traceable (AYC-918)

### PR DESCRIPTION
**Pull request by sirolad**
> [!NOTE]
> **Low Risk**
> Behavioral change is additive (callers can ignore the receipt); invite flow gains observability without altering auth or delivery failure handling beyond logging.
> 
> **Overview**
> Introduces a transport-neutral **`EmailDeliveryReceipt`** and changes **`EmailHandlerPort`** / **`SendEmailUseCase`** so successful sends return message ID, parsed SMTP status code, and accepted/rejected/pending recipients instead of `void`.
> 
> **`SmptHandler`** maps Nodemailer’s response into that receipt (normalizes recipient shapes, omits raw provider text, parses a leading 3-digit status when present). Unit tests cover receipt mapping and edge cases.
> 
> **`SendInvitationEmailUseCase`** is refactored into smaller helpers and logs structured **preparation** and **SMTP handover** events (message ID, status, recipient counts) for traceable invite delivery. Tests assert the new log payloads.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2d3dbc42408afaacb3da8daea8cdd5791d100ea7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by sirolad** [comment]
* **#1639** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1639?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1639?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: ayunis-core-backend/src/common/emails/infrastructure/handlers/smpt.handler.ts [hunk 1]
```diff
@@ -20,13 +22,24 @@ export class SmptHandler implements EmailHandlerPort {
...
       to: email.to,
       subject: email.subject,
       text: email.html ? '' : email.text,
       html: email.html,
     });
+    return {
+      messageId: receipt.messageId,
+      acceptedRecipients: receipt.accepted.map(recipientAddress),
+      rejectedRecipients: receipt.rejected.map(recipientAddress),
+      pendingRecipients: receipt.pending.map(recipientAddress),
```
[the remark this answers is not in the document]
Comment (sirolad) [reply ayunis-core-backend/src/common/emails/infrastructure/handlers/smpt.handler.ts]: Fixed in e0735a58. The SMTP receipt mapper now treats an omitted `pending` list as empty, with a regression test covering Mailpit/Nodemailer responses that omit the field.
